File: src/eShop.Admin/src/shared/types/product.ts
export interface Product {
    id: number;
    uuid: string;
    name: string;
    basePrice: number;
    shortDescription: string;
    mainImageUrl: string;
    isActive: boolean;
    categoryName: string;
    hasVariants: boolean;
    quantityInStock: number | null;
    createdDate: string;
    updatedDate: string | null;
    reviewCount: number;
    averageRating: number;
}

export interface ProductDetail {
    id: number;
    uuid: string;
    name: string;
    basePrice: number;
    description: string;
    shortDescription: string;
    isActive: boolean;
    isFeatured: boolean;
    categoryId: number;
    categoryName: string;
    hasVariants: boolean;
    quantityInStock: number | null;
    createdDate: string;
    updatedDate: string | null;
    images: ProductImage[];
    variants: ProductVariant[];
    attributes: ProductAttribute[];
    reviewCount: number;
    averageRating: number;
}

export interface ProductImage {
    id: number;
    url: string;
    altText: string;
    isMain: boolean;
    displayOrder: number;
}

export interface ProductVariant {
    id: number;
    name: string;
    price: number;
    quantityInStock: number | null;
    sku: string;
    isActive: boolean;
}

export interface ProductAttribute {
    id: number;
    attributeId: number;
    name: string;
    displayName: string;
    value: string;
}

File: src/eShop.Data/Entities/ProductAggregate/Product.cs
using eShop.Data.Entities.CategoryAggregate;
using eShop.Data.Entities.UserAggregate;
using Microsoft.EntityFrameworkCore;

namespace eShop.Data.Entities.ProductAggregate
{
    public class Product
    {
        public int Id { get; set; }
        public string Uuid { get; set; } = Guid.NewGuid().ToString();
        public required string Name { get; set; }
        [Precision(14, 2)]
        public required decimal BasePrice { get; set; }
        public required string Description { get; set; }
        public string? ShortDescription { get; set; }
        public bool IsActive { get; set; } = true;
        public int CategoryId { get; set; }
        public Category Category { get; set; } = null!;
        public bool HasVariants { get; set; }
        public int? QuantityInStock { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedDate { get; set; }
        public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
        public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();
        public ICollection<ProductAttribute> Attributes { get; set; } = new List<ProductAttribute>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
    }
}